ARM broke after upgrades

I'm not 100% where the issue lies because like an idiot I upgraded ARM and MakeMVK at the same time with out testing in between.

Previous to the upgrade discs were ripping fine.

I've attached the log. I'm not 100% certain if the issue is HandbreakCLI, MakeMKV or ARM itself.

ARM is seeing a disc has been inserted and is calling MakeMKV. MakeMKV appears to rip the disc successfully. The failure occurs at the transcoding portion.

I've re-compiled my HandbreakCLI just in case but that hasn't helped.

Log attached: 
[failedRip.log](https://github.com/ahnooie/automatic-ripping-machine/files/1497817/failedRip.log)

I'm thinking the issue is this:

```
[20:11:26] hb_scan: path=/mnt/ripTemp/raw//LOGICAL_VOLUME_ID_20171121_192805/CERTIFICATE, title_index=1
disc.c:352: error opening file BDMV/index.bdmv
disc.c:352: error opening file BDMV/BACKUP/index.bdmv
```

Some quick digging turned up that this might be a Handbreak issue?

That is part of Hand Brake scanning for files to transcode in that folder.
You can see further down it is not finding anything.

What files are in the raw ripped directory?
It spent 43 minutes ripping something, what files are there and what format?
On Wed, Nov 22, 2017 at 7:56 PM Eric <notifications@github.com> wrote:

> I'm not 100% where the issue lies because like an idiot I upgraded ARM and
> MakeMVK at the same time with out testing in between.
>
> Previous to the upgrade discs were ripping fine.
>
> I've attached the log. I'm not 100% certain if the issue is HandbreakCLI,
> MakeMKV or ARM itself.
>
> ARM is seeing a disc has been inserted and is calling MakeMKV. MakeMKV
> appears to rip the disc successfully. The failure occurs at the transcoding
> portion.
>
> I've re-compiled my HandbreakCLI just in case but that hasn't helped.
>
> Log attached:
> failedRip.log
> <https://github.com/ahnooie/automatic-ripping-machine/files/1497817/failedRip.log>
>
> I'm thinking the issue is this:
>
> [20:11:26] hb_scan: path=/mnt/ripTemp/raw//LOGICAL_VOLUME_ID_20171121_192805/CERTIFICATE, title_index=1
> disc.c:352: error opening file BDMV/index.bdmv
> disc.c:352: error opening file BDMV/BACKUP/index.bdmv
>
> Some quick digging turned up that this might be a Handbreak issue?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ahnooie/automatic-ripping-machine/issues/104>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AQVLdZapUuP8wHtrCM1EVCuYD6-QQIxRks5s5MLOgaJpZM4QoKvB>
> .
>

Hmm, I may not be much help, looks like you are using the backup rip
setting and mainfeature=true to do a straight makemkv rip and give it to
handbrake for processing. I have only ever used the mkv rip setting.

However, following the logic and comparing the logs, it looks like there is
definitely and issue that is preventing the following case in
video_transcode from being followed (and other if statements that rely on
the ID_CDROM_* variables to be present, they won’t be in the transcode step
unless we pass them as a parameter when we batch the transcode.

Your settings and this disc indicate you should be following this:

elif [ "$RIPMETHOD" = "backup" ] && [ "$MAINFEATURE" = true ] && [
"$ID_CDROM_MEDIA_BD" = "1" ]; then
echo "Transcoding BluRay main feature only." >> "$LOG"
# shellcheck disable=SC2086
$HANDBRAKE_CLI -i "$SRC" -o "$DEST/$LABEL.$DEST_EXT" --main-feature
--preset="$HB_PRESET" $HB_ARGS 2>> "$LOG"
rm -rf "$SRC"


Instead you are following this else statement:

else
echo "Transcoding all files." >> "$LOG"
# shellcheck disable=SC2045
for FILE in $(ls "$SRC")
do
filename=$(basename "$FILE")

#extension=${filename##*.}
filename=${filename%.*}

echo "Transcoding file $FILE" >> "$LOG"
# shellcheck disable=SC2086
$HANDBRAKE_CLI -i "$SRC/$FILE" -o "$DEST/$filename.$DEST_EXT"
--preset="$HB_PRESET" $HB_ARGS 2>> "$LOG"
rm "$SRC/$FILE"
done
rmdir "$SRC"


I had to make a similar change to pass the log file parameter to the
various other scripts, I can probably take a swing at a PR to address this
and other cases that may be missed by missing variables in the transcode
step.
On Wed, Nov 22, 2017 at 8:27 PM Eric <notifications@github.com> wrote:

> Sorry I already purged the files from that log I uploaded. I tried a
> second disc just to make sure it wasn't the problem.
>
> Here is a fresh log and the files:
> failedRip2.log
> <https://github.com/ahnooie/automatic-ripping-machine/files/1497849/failedRip2.log>
>
> ./BDROM_20171121_214107
> ./BDROM_20171121_214107/CERTIFICATE
> ./BDROM_20171121_214107/CERTIFICATE/BACKUP
> ./BDROM_20171121_214107/CERTIFICATE/BACKUP/id.bdmv
> ./BDROM_20171121_214107/CERTIFICATE/id.bdmv
> ./BDROM_20171121_214107/BDMV
> ./BDROM_20171121_214107/BDMV/BACKUP
> ./BDROM_20171121_214107/BDMV/BACKUP/CLIPINF
> ./BDROM_20171121_214107/BDMV/BACKUP/CLIPINF/00008.clpi
> ./BDROM_20171121_214107/BDMV/BACKUP/CLIPINF/00020.clpi
> ./BDROM_20171121_214107/BDMV/BACKUP/CLIPINF/00017.clpi
> ./BDROM_20171121_214107/BDMV/BACKUP/CLIPINF/00016.clpi
> ./BDROM_20171121_214107/BDMV/BACKUP/CLIPINF/00000.clpi
> ./BDROM_20171121_214107/BDMV/BACKUP/CLIPINF/00002.clpi
> ./BDROM_20171121_214107/BDMV/BACKUP/CLIPINF/00007.clpi
> ./BDROM_20171121_214107/BDMV/BACKUP/CLIPINF/00011.clpi
> ./BDROM_20171121_214107/BDMV/BACKUP/CLIPINF/00014.clpi
> ./BDROM_20171121_214107/BDMV/BACKUP/CLIPINF/00012.clpi
> ./BDROM_20171121_214107/BDMV/BACKUP/CLIPINF/00019.clpi
> ./BDROM_20171121_214107/BDMV/BACKUP/CLIPINF/00018.clpi
> ./BDROM_20171121_214107/BDMV/BACKUP/CLIPINF/00013.clpi
> ./BDROM_20171121_214107/BDMV/BACKUP/CLIPINF/00010.clpi
> ./BDROM_20171121_214107/BDMV/BACKUP/CLIPINF/00003.clpi
> ./BDROM_20171121_214107/BDMV/BACKUP/CLIPINF/00004.clpi
> ./BDROM_20171121_214107/BDMV/BACKUP/CLIPINF/00001.clpi
> ./BDROM_20171121_214107/BDMV/BACKUP/CLIPINF/00009.clpi
> ./BDROM_20171121_214107/BDMV/BACKUP/CLIPINF/00006.clpi
> ./BDROM_20171121_214107/BDMV/BACKUP/CLIPINF/00021.clpi
> ./BDROM_20171121_214107/BDMV/BACKUP/CLIPINF/00015.clpi
> ./BDROM_20171121_214107/BDMV/BACKUP/CLIPINF/00005.clpi
> ./BDROM_20171121_214107/BDMV/BACKUP/BDJO
> ./BDROM_20171121_214107/BDMV/BACKUP/MovieObject.bdmv
> ./BDROM_20171121_214107/BDMV/BACKUP/index.bdmv
> ./BDROM_20171121_214107/BDMV/BACKUP/PLAYLIST
> ./BDROM_20171121_214107/BDMV/BACKUP/PLAYLIST/00013.mpls
> ./BDROM_20171121_214107/BDMV/BACKUP/PLAYLIST/00009.mpls
> ./BDROM_20171121_214107/BDMV/BACKUP/PLAYLIST/00010.mpls
> ./BDROM_20171121_214107/BDMV/BACKUP/PLAYLIST/00014.mpls
> ./BDROM_20171121_214107/BDMV/BACKUP/PLAYLIST/00000.mpls
> ./BDROM_20171121_214107/BDMV/BACKUP/PLAYLIST/00007.mpls
> ./BDROM_20171121_214107/BDMV/BACKUP/PLAYLIST/00001.mpls
> ./BDROM_20171121_214107/BDMV/BACKUP/PLAYLIST/00011.mpls
> ./BDROM_20171121_214107/BDMV/BACKUP/PLAYLIST/00016.mpls
> ./BDROM_20171121_214107/BDMV/BACKUP/PLAYLIST/00012.mpls
> ./BDROM_20171121_214107/BDMV/BACKUP/PLAYLIST/00003.mpls
> ./BDROM_20171121_214107/BDMV/BACKUP/PLAYLIST/00005.mpls
> ./BDROM_20171121_214107/BDMV/BACKUP/PLAYLIST/00015.mpls
> ./BDROM_20171121_214107/BDMV/BACKUP/PLAYLIST/00006.mpls
> ./BDROM_20171121_214107/BDMV/BACKUP/PLAYLIST/00004.mpls
> ./BDROM_20171121_214107/BDMV/BACKUP/PLAYLIST/00002.mpls
> ./BDROM_20171121_214107/BDMV/BACKUP/PLAYLIST/00008.mpls
> ./BDROM_20171121_214107/BDMV/BACKUP/JAR
> ./BDROM_20171121_214107/BDMV/CLIPINF
> ./BDROM_20171121_214107/BDMV/CLIPINF/00008.clpi
> ./BDROM_20171121_214107/BDMV/CLIPINF/00020.clpi
> ./BDROM_20171121_214107/BDMV/CLIPINF/00017.clpi
> ./BDROM_20171121_214107/BDMV/CLIPINF/00016.clpi
> ./BDROM_20171121_214107/BDMV/CLIPINF/00000.clpi
> ./BDROM_20171121_214107/BDMV/CLIPINF/00002.clpi
> ./BDROM_20171121_214107/BDMV/CLIPINF/00007.clpi
> ./BDROM_20171121_214107/BDMV/CLIPINF/00011.clpi
> ./BDROM_20171121_214107/BDMV/CLIPINF/00014.clpi
> ./BDROM_20171121_214107/BDMV/CLIPINF/00012.clpi
> ./BDROM_20171121_214107/BDMV/CLIPINF/00019.clpi
> ./BDROM_20171121_214107/BDMV/CLIPINF/00018.clpi
> ./BDROM_20171121_214107/BDMV/CLIPINF/00013.clpi
> ./BDROM_20171121_214107/BDMV/CLIPINF/00010.clpi
> ./BDROM_20171121_214107/BDMV/CLIPINF/00003.clpi
> ./BDROM_20171121_214107/BDMV/CLIPINF/00004.clpi
> ./BDROM_20171121_214107/BDMV/CLIPINF/00001.clpi
> ./BDROM_20171121_214107/BDMV/CLIPINF/00009.clpi
> ./BDROM_20171121_214107/BDMV/CLIPINF/00006.clpi
> ./BDROM_20171121_214107/BDMV/CLIPINF/00021.clpi
> ./BDROM_20171121_214107/BDMV/CLIPINF/00015.clpi
> ./BDROM_20171121_214107/BDMV/CLIPINF/00005.clpi
> ./BDROM_20171121_214107/BDMV/META
> ./BDROM_20171121_214107/BDMV/STREAM
> ./BDROM_20171121_214107/BDMV/STREAM/00011.m2ts
> ./BDROM_20171121_214107/BDMV/STREAM/00013.m2ts
> ./BDROM_20171121_214107/BDMV/STREAM/00015.m2ts
> ./BDROM_20171121_214107/BDMV/STREAM/00003.m2ts
> ./BDROM_20171121_214107/BDMV/STREAM/00007.m2ts
> ./BDROM_20171121_214107/BDMV/STREAM/00001.m2ts
> ./BDROM_20171121_214107/BDMV/STREAM/00004.m2ts
> ./BDROM_20171121_214107/BDMV/STREAM/00017.m2ts
> ./BDROM_20171121_214107/BDMV/STREAM/00009.m2ts
> ./BDROM_20171121_214107/BDMV/STREAM/00012.m2ts
> ./BDROM_20171121_214107/BDMV/STREAM/00018.m2ts
> ./BDROM_20171121_214107/BDMV/STREAM/00010.m2ts
> ./BDROM_20171121_214107/BDMV/STREAM/00005.m2ts
> ./BDROM_20171121_214107/BDMV/STREAM/00006.m2ts
> ./BDROM_20171121_214107/BDMV/STREAM/00021.m2ts
> ./BDROM_20171121_214107/BDMV/STREAM/00014.m2ts
> ./BDROM_20171121_214107/BDMV/STREAM/00019.m2ts
> ./BDROM_20171121_214107/BDMV/STREAM/00020.m2ts
> ./BDROM_20171121_214107/BDMV/STREAM/00008.m2ts
> ./BDROM_20171121_214107/BDMV/STREAM/00000.m2ts
> ./BDROM_20171121_214107/BDMV/STREAM/00002.m2ts
> ./BDROM_20171121_214107/BDMV/STREAM/00016.m2ts
> ./BDROM_20171121_214107/BDMV/BDJO
> ./BDROM_20171121_214107/BDMV/MovieObject.bdmv
> ./BDROM_20171121_214107/BDMV/AUXDATA
> ./BDROM_20171121_214107/BDMV/index.bdmv
> ./BDROM_20171121_214107/BDMV/PLAYLIST
> ./BDROM_20171121_214107/BDMV/PLAYLIST/00013.mpls
> ./BDROM_20171121_214107/BDMV/PLAYLIST/00009.mpls
> ./BDROM_20171121_214107/BDMV/PLAYLIST/00010.mpls
> ./BDROM_20171121_214107/BDMV/PLAYLIST/00014.mpls
> ./BDROM_20171121_214107/BDMV/PLAYLIST/00000.mpls
> ./BDROM_20171121_214107/BDMV/PLAYLIST/00007.mpls
> ./BDROM_20171121_214107/BDMV/PLAYLIST/00001.mpls
> ./BDROM_20171121_214107/BDMV/PLAYLIST/00011.mpls
> ./BDROM_20171121_214107/BDMV/PLAYLIST/00016.mpls
> ./BDROM_20171121_214107/BDMV/PLAYLIST/00012.mpls
> ./BDROM_20171121_214107/BDMV/PLAYLIST/00003.mpls
> ./BDROM_20171121_214107/BDMV/PLAYLIST/00005.mpls
> ./BDROM_20171121_214107/BDMV/PLAYLIST/00015.mpls
> ./BDROM_20171121_214107/BDMV/PLAYLIST/00006.mpls
> ./BDROM_20171121_214107/BDMV/PLAYLIST/00004.mpls
> ./BDROM_20171121_214107/BDMV/PLAYLIST/00002.mpls
> ./BDROM_20171121_214107/BDMV/PLAYLIST/00008.mpls
> ./BDROM_20171121_214107/BDMV/JAR
> ./BDROM_20171121_214107/MAKEMKV
> ./BDROM_20171121_214107/MAKEMKV/AACS
> ./BDROM_20171121_214107/MAKEMKV/AACS/ContentRevocation.lst
> ./BDROM_20171121_214107/MAKEMKV/AACS/Content000.cer
> ./BDROM_20171121_214107/MAKEMKV/AACS/Unit_Key_RO.inf
> ./BDROM_20171121_214107/MAKEMKV/AACS/CPSUnit00001.cci
> ./BDROM_20171121_214107/MAKEMKV/AACS/MKB_RO.inf
> ./BDROM_20171121_214107/MAKEMKV/AACS/mcmf.xml
> ./BDROM_20171121_214107/MAKEMKV/AACS/ContentHash000.tbl
> ./BDROM_20171121_214107/MAKEMKV/AACS/MKB_RW.inf
> ./BDROM_20171121_214107/MAKEMKV/AACS/DUPLICATE
> ./BDROM_20171121_214107/MAKEMKV/AACS/DUPLICATE/ContentRevocation.lst
> ./BDROM_20171121_214107/MAKEMKV/AACS/DUPLICATE/Content000.cer
> ./BDROM_20171121_214107/MAKEMKV/AACS/DUPLICATE/Unit_Key_RO.inf
> ./BDROM_20171121_214107/MAKEMKV/AACS/DUPLICATE/CPSUnit00001.cci
> ./BDROM_20171121_214107/MAKEMKV/AACS/DUPLICATE/MKB_RO.inf
> ./BDROM_20171121_214107/MAKEMKV/AACS/DUPLICATE/mcmf.xml
> ./BDROM_20171121_214107/MAKEMKV/AACS/DUPLICATE/ContentHash000.tbl
> ./BDROM_20171121_214107/MAKEMKV/AACS/DUPLICATE/MKB_RW.inf
> ./BDROM_20171121_214107/MAKEMKV/discattd.dat
> ./BDROM_20171121_214107/MAKEMKV/CMAP
> ./BDROM_20171121_214107/MAKEMKV/CMAP/00017.cmap
> ./BDROM_20171121_214107/MAKEMKV/CMAP/00008.cmap
> ./BDROM_20171121_214107/MAKEMKV/CMAP/00003.cmap
> ./BDROM_20171121_214107/MAKEMKV/CMAP/00020.cmap
> ./BDROM_20171121_214107/MAKEMKV/CMAP/00018.cmap
> ./BDROM_20171121_214107/MAKEMKV/CMAP/00012.cmap
> ./BDROM_20171121_214107/MAKEMKV/CMAP/00011.cmap
> ./BDROM_20171121_214107/MAKEMKV/CMAP/00009.cmap
> ./BDROM_20171121_214107/MAKEMKV/CMAP/00006.cmap
> ./BDROM_20171121_214107/MAKEMKV/CMAP/00000.cmap
> ./BDROM_20171121_214107/MAKEMKV/CMAP/00005.cmap
> ./BDROM_20171121_214107/MAKEMKV/CMAP/00015.cmap
> ./BDROM_20171121_214107/MAKEMKV/CMAP/00001.cmap
> ./BDROM_20171121_214107/MAKEMKV/CMAP/00021.cmap
> ./BDROM_20171121_214107/MAKEMKV/CMAP/00010.cmap
> ./BDROM_20171121_214107/MAKEMKV/CMAP/00019.cmap
> ./BDROM_20171121_214107/MAKEMKV/CMAP/00004.cmap
> ./BDROM_20171121_214107/MAKEMKV/CMAP/00007.cmap
> ./BDROM_20171121_214107/MAKEMKV/CMAP/00002.cmap
> ./BDROM_20171121_214107/MAKEMKV/CMAP/00016.cmap
> ./BDROM_20171121_214107/MAKEMKV/CMAP/00014.cmap
> ./BDROM_20171121_214107/MAKEMKV/CMAP/00013.cmap
>
> And the directory size
>
> root@ripbox:/mnt/ripTemp/raw# du -hs BDROM_20171121_214107/
> 18G     BDROM_20171121_214107/
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/ahnooie/automatic-ripping-machine/issues/104#issuecomment-346514231>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AQVLdUsXwBjrL2IhOOzlFjtzzfUI3ciIks5s5Mn4gaJpZM4QoKvB>
> .
>

--- a/video_rip.sh
+++ b/video_rip.sh
@@ -4,8 +4,8 @@
 # shellcheck source=config
 # shellcheck disable=SC1091
 source "$ARM_CONFIG"
-# shellcheck disable=SC1090
-while read info_var; do export $info_var; done < "${DISC_INFO}"
+# shellcheck disable=SC2163
+while read -r info_var; do export $info_var; done < "${DISC_INFO}"
 
 VIDEO_TITLE=$1
 HAS_NICE_TITLE=$2

--- a/video_rip.sh
+++ b/video_rip.sh
@@ -5,7 +5,7 @@
 # shellcheck disable=SC1091
 source "$ARM_CONFIG"
 # shellcheck disable=SC2163
-while read -r info_var; do export $info_var; done < "${DISC_INFO}"
+while read -r info_var; do export "${info_var}"; done < "${DISC_INFO}"
 
 VIDEO_TITLE=$1
 HAS_NICE_TITLE=$2

--- a/video_rip.sh
+++ b/video_rip.sh
@@ -5,7 +5,7 @@
 # shellcheck disable=SC1091
 source "$ARM_CONFIG"
 # shellcheck disable=SC1090
-source "$DISC_INFO"
+while read info_var; do export $info_var; done < "${DISC_INFO}"
 
 VIDEO_TITLE=$1
 HAS_NICE_TITLE=$2

